src/lib/cimb_translator/CimbReader.cpp

```diff
@@ -141,6 +148,114 @@ bool CimbReader::done() const
 	return !_good or _positions.done();
 }
 
+void CimbReader::init_ccm(unsigned color_bits, unsigned interleave_blocks, unsigned interleave_partitions, unsigned fountain_blocks)
```

Comment:
This method coordinates the new color correction logic. It needs a refactor, but the comments hopefully explain the basics (cimbar-py also contains an implementation).

Reply:
But to simplify:
* we use the deterministic 48-bit fountain-chunk header to sample expected color values to use as an ad-hoc color "key".
* the sampled color averages for the 4 colors are then [transformed](https://en.wikipedia.org/wiki/Moore-Penrose_inverse) into the ccm